clang/test/CIR/CodeGen/class_cast.cpp

```diff
@@ -0,0 +1,45 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -O2 -fclangir -emit-cir %s -o - 2>&1 | FileCheck %s --check-prefix=CIR
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -O2 -fclangir -emit-llvm %s -o - 2>&1 | FileCheck %s --check-prefix=LLVM
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -O2 -emit-llvm %s -o - 2>&1 | FileCheck %s --check-prefix=OGCG
+class Base {
+  // CIR-LABEL: _ZN4BaseaSERKS_
+  // CIR-SAME: [[ARG0:%.*]]: !cir.ptr{{.*}}, [[ARG1:%.*]]: !cir.ptr{{.*}}
+  // CIR: [[ALLOCA_0:%.*]] =  cir.alloca
+  // CIR: [[ALLOCA_1:%.*]] =  cir.alloca
+  // CIR: [[ALLOCA_2:%.*]] =  cir.alloca
+  // CIR: cir.store [[ARG0]], [[ALLOCA_0]]
+  // CIR: cir.store [[ARG1]], [[ALLOCA_1]]
+  // CIR: [[LD_0:%.*]] = cir.load deref [[ALLOCA_0]]
+  // CIR: cir.store align(8) [[LD_0]], [[ALLOCA_2]]
+  // CIR: [[LD_1:%.*]] = cir.load [[ALLOCA_2]]
+  // CIR: cir.return [[LD_1]]
+};
+
+class Derived : Base {
+  Derived &operator=(Derived &);
+};
+Derived &Derived::operator=(Derived &B) {
+  // CIR-LABEL: _ZN7DerivedaSERS_
+  // CIR-SAME: [[ARG0:%.*]]: !cir.ptr{{.*}}, [[ARG1:%.*]]: !cir.ptr{{.*}}
+  // CIR: [[ALLOCA_0:%.*]] =  cir.alloca
+  // CIR: [[ALLOCA_1:%.*]] =  cir.alloca
+  // CIR: [[ALLOCA_2:%.*]] =  cir.alloca
```

Comment:
Please clean up the checks a bit in terms of what matters most for the sake of this change?